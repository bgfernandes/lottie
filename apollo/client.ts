
import { ApolloClient, InMemoryCache, NormalizedCacheObject } from '@apollo/client'

let apolloClient: ApolloClient<NormalizedCacheObject>

if (typeof window === 'undefined') {
  // code is running in the server side
  const { SchemaLink } = require('@apollo/client/link/schema')
  const { schema } = require('./schema')

  apolloClient = new ApolloClient({
    ssrMode: true,
    link: new SchemaLink({ schema }),
    cache: new InMemoryCache(),
  })
} else {
  // code is running in the browser
  const { HttpLink } = require('@apollo/client/link/http')

  apolloClient = new ApolloClient({
    ssrMode: false,
    link: new HttpLink({
      uri: '/api/graphql',
      credentials: 'same-origin',
    }),
    cache: new InMemoryCache(),
  })
}

export default apolloClient
