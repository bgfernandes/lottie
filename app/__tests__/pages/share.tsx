/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react'
import { MockedProvider } from '@apollo/client/testing'
import Share from '../../pages/share'

describe('Share', () => {
  it('renders successfully', () => {
    const { container } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <Share />
      </MockedProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
