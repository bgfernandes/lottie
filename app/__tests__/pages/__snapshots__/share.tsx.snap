// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share renders successfully 1`] = `
<div>
  <div>
    Choose a file to upload
    <br />
    <input
      required=""
      type="file"
    />
  </div>
</div>
`;
